### ISSUE TITLE
User Can't Login After Applying Turpentine VCL

### ISSUE THREAD

**Issue by sabonmichal** [nexcess/magento-turpentine#642]
Turpentine dramatically increases server performance as evidenced by magespeedtest tool but we have one small problem: orders are not coming through because when customer gets to checkout (checkout/onepage/ page) and tries to login, they get redirected back to the login screen (the browser refreshes to the same page instead of going to the next step in checkout).  We are using ssl on ubuntu/nginx server.

This issue does is not present when using varnish without applying turpentine however cached content and server performance are reduced.

Any suggestions are appreciated.

Thank you :)

**Comment by theycallmepepper** [comment]
Is it across every browser?  I had a similar issue at one point that I believe was corrected with ```user_agent_normalization``` and/or ```normalize-encoding``` configuration.

**Comment by aricwatson** [comment]
Additionally, you could try black-listing your checkout URLs to ensure that varnish is bypassed for those - if you haven't found a solution yet. 

If you're still having issues feel free to reply and I'll re-open the issue.

**Comment by adam105** [comment]
ok, thanks. We gave up on Turpentine extension for now, hours of effort but
couldn't get it to work, might try again in the future.

**Comment by beckindesigns** [comment]
Here is what worked for me and I don't see this answer anywhere.

In my case it was a turpentine setting. I used the backdoor to get into the admin area by going to the downloader page (yoursite.com/downloader) and logging in. After that, I changed **_System > Configurations > Turpentine > Varnish Options > General Options > Use VCL fix_** to "disabled".

"When Enable is selected, a VCL fix will be used to prevent formKey issues. If Disable is selected, an observer will be used."

It appears the observer did the trick. Also, I am not using an override of the `app/code/local/Mage/Core/Model/Session.php` file.
